CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+
+- Our [lack of line numbers is a feature](https://highlightjs.readthedocs.io/en/latest/line-numbers.html). It's usually pretty simple to add them yourself ([example](https://github.com/taufik-nurrohman/highlight.ln.js)).
+- We [embrace plug-ins and extensions](https://github.com/highlightjs/highlight.js/issues/2225) rather than adding lots of tiny features and configuration options.
+- We are much more than a keyword highlighter. We do make attempts to understand context...
+- ...Yet we are not a full parser for any language, we're still just a fancy nested pattern matcher.
+- Our [auto-detect is sadly not powered by magic unicorn dust](https://github.com/highlightjs/highlight.js/issues/1213). It's best effort, not best in class.
+
+
+## Feature Requests
+Feature requests are always welcome but many times the answer is:
+
+> That's a great idea, you should write a plug-in for that! It doesn't belong in the core library and it's probably something you could do with only a small amount of code.
+
+If you'd like to contribute a feature you should always first open an issue to discuss if the feature belongs in core vs a plug-in.  Often this is a great way to get tips for implemention as well or links to prior disussions with context, etc.
+
+
+## Reporting an Issue
+(Misc.)
+- If you found a language highlighting issue, use [this JSFiddle](https://jsfiddle.net/ajoshguy/2bmdswn6/) to create a test case.
+
+
+## Fixing an Issue (PRs)
+If you're new to contributing to open-source, have a look at [this GitHub Guide](https://guides.github.com/activities/forking). It explains the general process of GitHub collaboration.
+
+If you feel comfortable with our [prerequisites](#prerequisites), you can grab any issue [marked `beginner friendly`](https://github.com/highlightjs/highlight.js/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22).
+
+
+### Rules
+Don't worry; these rules are pretty standard.
+
+Please open an issue before your PR, so we can explore and discuss the topic at hand. Your time is important, and we need to be sure it's well-spent.
+
+**Before** you start coding, keep these tips in mind:
+- You should usually add markup tests (ie. whenever you've made a significant grammar change or fixed a bug)
+- Change only what needs to be changed; don't re-lint/rewrite whole files when fixing bugs
+- Linting needs a dedicated commit
+
+**After** you make your changes, we have some housekeeping tasks for you. See the PR template for details.
+
+
+## Setup and Run
+
+You'll need [Node.js](https://nodejs.dev/download/), and optionally [yarn](https://yarnpkg.com/).
+
+Often when contributing a PR (that doesn't make any specific changes to browser
+features) it's more than sufficient to build and test only the Node.js build.
+Our CI process will guarantee that the browser build is still green.
+
+```console
+npm run build
+npm run test
+```
+
+The browser library must be built and tested separately:
+
+```console
+npm run build-browser
```

Comment:
`build_and_test_browser` ?

Reply:
This should be consistent with whatever we do above.